sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditionsTest.java

```diff
@@ -272,57 +333,318 @@ void testAssociateNewLocationForHandle() {
+        assertEquals(ContextAssociation.ASSOC, assocCaptor.getValue());
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private void sleep(final long delayInMillis) {
+        final long start = System.currentTimeMillis();
+        final long end = start + delayInMillis;
+        long now = System.currentTimeMillis();
+        while (now < end) {
+            try {
+                Thread.sleep(end - now);
+            } catch (InterruptedException e) {
+                // do nothing
+            }
+            now = System.currentTimeMillis();
+        }
+    }
+
+    @Test
+    @DisplayName(
+            "associateNewLocations: EpisodicContextReport is too slow and the Association is hence not yet udated.")
+    void testAssociateNewLocationForHandleReportTooSlow() {
+        updated = false;
+        associateNewLocationsSetup();
+
+        assertFalse(
+                ManipulationPreconditions.AssociateLocationsManipulation.manipulation(injector),
+                "Manipulation should've failed");
+        assertTrue(testRunObserver.isInvalid(), "Test run should have been invalid.");
+    }
+
     @Test
     @DisplayName("associateNewLocations: New location not associated")
     void testAssociateNewLocationForHandleWrongAssociation() {
         associateNewLocationsSetup();
+        final List<MdibAccessObserver> observers = new ArrayList<>();
 
         // introduce error, context won't be associated
         when(mockLocationContextState.getContextAssociation()).thenReturn(ContextAssociation.DIS);
 
+        final MdibAccessObservable mockMdibAccessObservable = mock(MdibAccessObservable.class);
+        when(mockTestClient.getSdcRemoteDevice().getMdibAccessObservable()).thenReturn(mockMdibAccessObservable);
+        final AtomicReference<MdibAccessObserver> observer = new AtomicReference<>();
+        final Object signal = new Object();
+        doAnswer((arguments) -> {
+                    synchronized (signal) {
+                        observer.set(arguments.getArgument(0));
+                        signal.notifyAll();
+                    }
+                    return null;
+                })
+                .when(mockMdibAccessObservable)
+                .registerObserver(any());
+
+        final Thread t = new Thread(() -> {
+            ManipulationPreconditions.AssociateLocationsManipulation.EpisodicContextReportStateHandleObserver
+                    mdibAccessObserver;
+            for (int i = 0; i < 2; i++) {
+                mdibAccessObserver =
+                        (ManipulationPreconditions.AssociateLocationsManipulation
+                                        .EpisodicContextReportStateHandleObserver)
+                                observer.get();
+                synchronized (signal) {
+                    while (mdibAccessObserver == null) {
+                        try {
+                            signal.wait(300);
+                        } catch (InterruptedException e) {
```

Comment:
do not catch it here if you are not reacting to it

also you are not even interrupting the thread from another thread, thus there is no good way of reacting to it and it should just be escalated

Reply:
same as above.

Reply:
Improved just like above.